SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -223,9 +226,98 @@ void USpatialWorkerConnection::ConnectToLocator()
+
+	Worker_Alpha_LocatorParameters LocatorParams = {};
+	LocatorParams.player_identity.player_identity_token = PlayerIdentityTokenCStr.Get();
+	LocatorParams.player_identity.login_token = LoginTokenCStr.Get();
+
+	WorkerLocator = Worker_Alpha_Locator_Create(TCHAR_TO_UTF8(*LocatorConfig.LocatorHost), 0, &LocatorParams);
+
+	// TODO: Move creation of connection parameters into a function somehow
+	Worker_ConnectionParameters ConnectionParams = Worker_DefaultConnectionParameters();
+	FTCHARToUTF8 WorkerTypeCStr(*LocatorConfig.WorkerType);
+	ConnectionParams.worker_type = WorkerTypeCStr.Get();
+	ConnectionParams.enable_protocol_logging_at_startup = LocatorConfig.EnableProtocolLoggingAtStartup;
+
+	Worker_ComponentVtable DefaultVtable = {};
+	ConnectionParams.component_vtable_count = 0;
+	ConnectionParams.default_component_vtable = &DefaultVtable;
+
+	ConnectionParams.network.connection_type = LocatorConfig.LinkProtocol;
+	ConnectionParams.network.use_external_ip = LocatorConfig.UseExternalIp;
+
+	FString ProtocolLogDir = FPaths::ConvertRelativePathToFull(FPaths::ProjectLogDir()) + TEXT("protocol-log-");
+#if PLATFORM_WINDOWS == 1 || PLATFORM_XBOXONE == 1
+	ConnectionParams.enable_protocol_logging_at_startup = true;
+#if PLATFORM_XBOXONE == 1
+	ConnectionParams.protocol_logging.log_prefix = "T:\\protocol-log-";
+#else
+	ConnectionParams.protocol_logging.log_prefix = TCHAR_TO_UTF8(*ProtocolLogDir);
+#endif
+	UE_LOG(LogSpatialWorkerConnection, Log, TEXT("Protocol log prefix: %s"), UTF8_TO_TCHAR(ConnectionParams.protocol_logging.log_prefix));
+#endif
+	// end TODO
+
+	Worker_ConnectionFuture* ConnectionFuture = Worker_Alpha_Locator_ConnectAsync(WorkerLocator, &ConnectionParams);
+
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [ConnectionFuture, this]
+	{
+		WorkerConnection = Worker_ConnectionFuture_Get(ConnectionFuture, nullptr);
+
+		Worker_ConnectionFuture_Destroy(ConnectionFuture);
+		if (Worker_Connection_IsConnected(WorkerConnection))
+		{
+			AsyncTask(ENamedThreads::GameThread, [this]
+			{
+				this->bIsConnected = true;
+				this->OnConnected.ExecuteIfBound();
+			});
+		}
+		else
+		{
+			Worker_OpList* OpList = Worker_Connection_GetOpList(WorkerConnection, 0);
+			for (int i = 0; i < (int)OpList->op_count; i++)
+			{
+				if (OpList->ops[i].op_type == WORKER_OP_TYPE_DISCONNECT)
+				{
+					UE_LOG(LogSpatialWorkerConnection, Error, TEXT("Couldn't connect to SpatialOS: %s"), UTF8_TO_TCHAR(OpList->ops[i].disconnect.reason));
+					GEngine->AddOnScreenDebugMessage(-1, 200.0f, FColor::Red, FString::Printf(TEXT("Couldn't connect to SpatialOS: %s"), UTF8_TO_TCHAR(OpList->ops[i].disconnect.reason)));
+				}
+			}
+
+			// TODO: Try to reconnect - UNR-576
+		}
+	});
+}
+
+SpatialConnectionType USpatialWorkerConnection::GetConnectionType() const
 {
-	return !LocatorConfig.LoginToken.IsEmpty();
+	if (!LocatorConfig.LoginToken.IsEmpty())
+	{
+		return SpatialConnectionType::Locator;
+	}
+	else if (!LegacyLocatorConfig.LoginToken.IsEmpty())
```

Comment:
This code path will never be hit based on CLI arguments alone, since the command-line argument that the legacy config searches for ("loginToken") is exactly the same. Might be worth checking for the player identity token above instead.

Reply:
I've swapped this to check playerIdentityToken